packages/core/src/types.ts

```diff
@@ -272,7 +272,15 @@ export type Composition = {
   componentSettings?: ExperienceComponentSettings;
 };
 
-export type DesignTokensDefinition = { [key: string]: string | DesignTokensDefinition };
+export type DesignTokensDefinition = {
+  spacing?: Record<string, string>;
+  sizing?: Record<string, string>;
+  colors?: Record<string, string>;
+  borders?: Record<string, { size: string; color: string }>;
+  fontSize?: Record<string, string>;
+} & {
+  [key: string]: string | Record<string, string>;
```

Comment:
We may want to make this a recursive type so that there can be multiple nestings. Maybe we can make this change in the future so either adding a comment or actually making a recursive type here should suffice